### PR TITLE
api/controller

### PR DESCRIPTION
**Pull request by Courtcircuits**
This PR aims to specify the OpenAPI v3 for the REST API interface of the controller but also all protobuf files for our gRPC interface.

### CODE REVIEW

File: api/openapi/controller/controller.openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
+openapi: 3.0.1
+info:
+  title: SealCI
+  description: ''
+  version: 1.0.0
```
Comment (sameo) [inline api/openapi/controller/controller.openapi.yaml:5]:
```suggestion
  version: 0.0.1
```

;-)

**Review by sameo — COMMENTED** [review]
Looks pretty good. Simple and to the point. I only have a few comments.

File: api/openapi/controller/controller.openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+          description: ''
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Success
+          content:
+            application/json:
+              schema: *ref_0
```
[the remark this answers is not in the document]
Comment (Courtcircuits) [reply api/openapi/controller/controller.openapi.yaml:105]: ref_0 is defined at line 82 with the &ref_0 notation. Thus, it references '#/components/schemas/pipeline'. Basically this syntax avoid references duplication. Maybe not the most readable syntax but it is compatible with swagger

File: proto/action.proto [hunk 3]
```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package action;
+
+service action {
```
Comment (sameo) [inline proto/action.proto:5]: Here you want to define a service for scheduling an action:

- It is a scheduler service and should be defined as part of the scheduler API
- Its name should describe the service that it exposes, e.g. `SchedulerService`

File: api/openapi/controller/controller.openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+          schema:
+            type: boolean
+      responses:
+        '200':
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: array
+                items: &ref_0
```
Comment (sameo) [inline api/openapi/controller/controller.openapi.yaml:82]:
```suggestion
                items: &pipeline_status
```

**Review by sameo — APPROVED** [review]

**Pull request merged**